src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -3985,8 +3985,12 @@ MergeTreeData::PartsToRemoveFromZooKeeper MergeTreeData::removePartsInRangeFromW
     /// FIXME refactor removePartsFromWorkingSet(...), do not remove parts twice
     removePartsFromWorkingSet(txn, parts_to_remove, clear_without_timeout, lock);
 
+    /// We can only create a covering part for a blocks range that starts with 0 (otherwise we may get "intersecting parts").
+    /// Maybe we could do it by incrementing mutation version to get a name for the empty covering part,
+    /// but it's okay to simple avoid creating it for DROP PART
+    bool is_drop_part = drop_range.min_block;
```

Comment:
This is difficult to understand.

Here, I can read that "the range means 'drop part' if the min block number is not zero" which contradicts the comment.

Moreover, it's unclear when we have a zero block number. I thought block numbers start from one.

Reply:
> "the range means 'drop part' if the min block number is not zero" which contradicts the comment

Yes, that's correct, but it does not contradict the comment. What do you find contradictory? 

> Also, why it's "drop part", not "drop partition"?

Because we have a problem with "drop part", not with "drop partition"

> Moreover, it's unclear when we have a zero block number. I thought block numbers start from one.

Block numbers start from 1 for MergeTree. For ReplicatedMergeTree they start from 0. So we have 0 for the first block in partition, and for all DROP_RANGEs for the whole partition

Reply:
> Block numbers start from 1 for MergeTree. For ReplicatedMergeTree they start from 0.

This is non obvious and has to be clarified in comments.
Would you mind also changing the first block number to 0 for non-replicated MergeTree?

> What do you find contradictory?

If we drop a part with block number zero, it will be a "drop part" operation, but `bool is_drop_part` will be false.

Reply:
> Would you mind also changing the first block number to 0 for non-replicated MergeTree?

I can change it in a separate PR, but it will require updating a few hundred tests...

> If we drop a part with block number zero, it will be a "drop part" operation, but bool is_drop_part will be false.

Yes, you're right. But it's not a problem because we don't have any parts on the left of the range, so we don't need to distinguish "drop part" in this case. I will update the comment and change the flag name